indico/modules/events/abstracts/templates/management/_notification_tpl_list.html

```diff
@@ -1,60 +1,112 @@
+                                                   title="{% trans %}Edit e-mail text{% endtrans %}"
+                                                   data-title="{% trans %}Edit e-mail text{% endtrans %}"
+                                                   data-ajax-dialog
+                                                   data-update="#email-template-manager">
+                                                   {% trans %}E-mail text{% endtrans %}
+                                                </a>
+                                            </li>
+                                        </ul>
+                                        <a href="#" class="i-button js-delete-item icon-remove icon-only"
+                                           title="{% trans %}Delete template{% endtrans %}"
+                                           data-qtip-style="danger"
+                                           data-href="{{ url_for('.email_tpl_delete', tpl) }}"
+                                           data-method="DELETE"
+                                           data-confirm="{% trans %}Are you sure you want to delete this template?{% endtrans %}"
+                                           data-title="{% trans %}Delete template{% endtrans %}"
+                                           data-update="#email-template-manager">
                                         </a>
-                                    </li>
-                                    <li>
-                                        <a data-href="{{ url_for('.email_tpl_text_edit', tpl) }}"
-                                           title="{% trans %}Edit e-mail text{% endtrans %}"
-                                           data-title="{% trans %}Edit e-mail text{% endtrans %}"
-                                           data-ajax-dialog>
-                                           {% trans %}E-mail text{% endtrans %}
-                                        </a>
-                                    </li>
+                                    </div>
+                                </div>
+                            </div>
+                            <div class="i-box-content">
+                                <div>
+                                    {% trans %}This email will be sent when the abstract is:{% endtrans %}
+                                </div>
+                                <ul>
+                                    {% for rule in tpl.rules if rule %}
+                                        <li class="notification-rule">
+                                            <strong>{{ states[rule['state'][0]] }}</strong>
+                                            {% if 'track' in rule %}
+                                                {% trans %}in{% endtrans %}
+                                                <strong>
+                                                    {% if rule['track'][0] == '*' %}
```

Comment:
Yes - when using `foo.bar` it first tries `foo.bar` and if that raises an AttributeError it tries `foo['bar']`. If you do `foo['bar']` it does it exactly the other way round.

So for static keys you can always uses the dot notation in Jinja; only if you either need a key that also exists as an attribute (e.g. `somedict['items']`) or with a dynamic value (`somedict[somekey]`) you need the normal bracket notation.